docs/scos/dev/upgradability-services/new-documentation/core-method-used-on-project-level.md

```diff
@@ -0,0 +1,119 @@
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+Modules have public and private APIs. While public API updates always support backward compatibility, private API updates can break backward compatibility. So, backward compatibility is not guaranteed in the private API. For example, if you use a core method on the project level, and it is updated or removed, it can cause unexpected issues during updates.
+
+For more information about module APIs, see [Definition of Module API](/docs/scos/dev/architecture/module-api/definition-of-module-api.html).
+
+## Example of code that causes the upgradability error
+
+`CustomerFacade` uses the `createCustomerReader` and `getCustomerCollection`. This is a  private API.
+
+```php
+namespace Pyz\Zed\Customer\Business;
+
+use Spryker\Zed\Customer\Business\CustomerFacade as SprykerCustomerFacade;
+
+/**
+ * @method \Pyz\Zed\Customer\Business\CustomerBusinessFactory getFactory()
+ */
+class CustomerFacade extends SprykerCustomerFacade implements CustomerFacadeInterface
+{
+   /**
+     * ...
+     */
+    public function getCustomerCollection(...): ...
+    {
+        return $this->getFactory()
+            ->createCustomerReader()
+            ->getCustomerCollection(...);
+    }
+}
+```
+
+## Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Pyz\Zed\Customer\Business\CustomerFacade
+"PrivateApi:Persistence Please avoid Spryker dependency: "{name}""
+************************************************************************************************************************
+```
+
+## Example of resolving the Evaluator check error
+
+To resolve the error provided in the example, do the following steps:
+1. Recommended: Check if it's possible to extend the functionality using the [Configuration strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#configuration).
+2. Recommended: Check if it's possible to extend the functionality using the [Plug and Play strategy](/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#plug-and-play).
+3. Recommended: Check if it is possible to extend functionality with "Project Modules" strategy (link to "Project Modules") - https://docs.spryker.com/docs/scos/dev/back-end-development/extending-spryker/development-strategies/development-strategies.html#project-modules
+4. Not recommended: Replace the private API core functionality with their copies on the project level and use a unique naming for it:
+    1. Not recommended: Add new unique method in the factory to fetch the business model.
+    2. Not Recommended: Add new unique method in business model.
+
+{% info_block infoBox "" %}
+We are working on introducing a way to report such cases and add more extension points in the core.
```

Comment:
Do you mean add more extension points based on the reported cases or regardless of that (in general)?